src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarHeader.Read.cs

```diff
@@ -311,6 +319,8 @@ private async Task ProcessDataBlockAsync(Stream archiveStream, bool copyData, Ca
             {
                 MemoryStream copiedData = new MemoryStream();
                 TarHelpers.CopyBytes(archiveStream, copiedData, _size);
+                // Reset position pointer so the user can do the first DataStream read from the beginning
```

Comment:
nit: slightly better wording (don't trigger CI for this suggestion)

```suggestion
                // Reset stream position so the user can do the first DataStream read from the beginning
```

also: do other usages of `TarHelpers.CopyBytes` also need to reset the position? If so, we could move it to this method itself.